Bugfix SS pin definitions

Hasi123 opened the pull request:
Strangely the old code worked on some boards, but not on others. Found this little bug and thought I would create a pull request.

Hasi123 committed:
Bugfix SS pin definitions

prunkdump commented:
Hi Hasi !

Does you understand how to identify the different board type ?

You talk about Pro-Mini only ?

Hasi123 commented:
Well you already defined B0 in the code, and for that the port register is PORTB and the ddr is DDRB.
```
#ifndef SD_SS_PORT
#define SD_SS_PORT B
#endif
/* !!! default changed !!! */
#ifndef SD_SS_PIN
#define SD_SS_PIN 0
#endif
```

I do not understand your question, the SD SS pin is not board dependent, but can be freely chosen depending on where you connect the SS pin of the SD card.

Hasi123 committed:
Add binaries

Hasi123 committed:
fix: skip non-data HEX record types (0x02-0x05) in SD card loader

The bootloader's hex parser did not handle Intel HEX record types
other than 0x00 (Data) and 0x01 (EOF). Record type 0x04 (Extended
Linear Address) was being treated as data, causing a 2-byte shift
in the firmware written to flash.

Fix: skip writing to flash for any record type != 0x00.
Also disabled LED_START_FLASHES (GNUVario has no status LED)
to keep bootloader under 2048 bytes.

Bootloader size: 2012 bytes (was 2034)
